tests/core/test_socks_proxy.py

```
@@ -0,0 +1,282 @@
+        mock_socks.SOCKS4 = 1
+        mock_socks.socksocket = MagicMock()
+        
+        with patch.dict('sys.modules', {'socks': mock_socks}):
+            set_socks_proxy("socks5://user:pass@proxy.example.com:1080")
+            
+            mock_socks.set_default_proxy.assert_called_once_with(
+                2,  # SOCKS5
+                "proxy.example.com",
+                1080,
+                username="user",
+                password="pass"
```

Ruff (0.14.14)

[error] Possible hardcoded password assigned to argument: "password"

(S106)